lib/models/ui/ranking/ranking_details.dart

```diff
@@ -0,0 +1,37 @@
+import "package:flutter/foundation.dart";
+import "package:proxima/models/ui/ranking/ranking_element_details.dart";
+
+/// A class that stores data for the ranking widget.
+/// Elements in [rankElementDetailsList] should have non-null [userRank].
+///
+/// Note that we don't ensure that the [userRank] are not empty as we want to leave flexibility
+/// to the developer. Moreover, when used in the UI, this will yield a runtime error
+/// if the [userRank] is null.
+@immutable
+class RankingDetails {
+  const RankingDetails({
+    required this.userRankElementDetails,
+    required this.rankElementDetailsList,
+  });
+
+  /// Details of the user's rank.
```

Comment:
Can you add here (in the documentation) the fact that the `userRank` attribute of this `RankingElementDetails` will be null.

Reply:
You are completely right !

Added in: [8a55765](https://github.com/ProximaEPFL/proxima/pull/319/commits/8a55765d5958a96c8109fae1ca8da60b472f64cf)

Thank you !